### DISCUSSION TITLE
Schalter - Lang und Kurz

### DISCUSSION THREAD

**Discussion by dibadodo** [thkl/hap-homematic#307] [General]
Hallo,

Wie kann ich denn einen Schalter einbinden, der unterschiedliche Aktionen bei einem langen und bei einem kurzen Schaltersignal / Betätigen ausführt. In Homematic habe ich einen 6-fach Schalter, bei dem z.B. der Schalter 5 bei kurzen betätigen das Licht an macht und bei langen betätigen wieder aus. Ich kann zwar beim Einbinden des Schalters in HAP Hoheit festlegen ob der simulierte Tastendruck kurz oder lang sein soll, aber wenn ich den Schalter beim ersten anlegen z.B. mit Lang eingefügt habe und ihn dann ein zweites mal, jetzt mit kurz, einbinden will, lässt das das System nicht zu, da der Schalter schon vorhanden ist.
Wie kann ich das einbinden?
Über ein Programm geht das zwar, aber dann sehe ich den Status nicht. Ich weiß aus nie, ob das Licht jetzt an oder aus ist. ;)

Danke 

VG

**Comment by thkl** [comment]
Gute Frage, das geht nicht. Dazu müsste ich den Service so umbauen, das man ggf 2 Switch Knöpfe hat.

**Comment by thkl** [comment]
![image](https://user-images.githubusercontent.com/884355/103377798-a8335c00-4ae0-11eb-9907-0b5b1b582eca.png)

wird in Zukunft zu:

![image](https://user-images.githubusercontent.com/884355/103377815-b84b3b80-4ae0-11eb-8c96-5c6ce6f08331.png)

**Comment by dibadodo** [comment]
Wenn das dann auch zwei verschiedene Aktionen sind, die man damit schalten kann, wäre das in Ordnung. So rein von der Grafik sieht mir das eher so aus als wenn das dann im Endeffekt zwei Schalter für eine Aktion sind. Aber es ist ja eher ein Schalter für zwei (verschiedene) Aktionen.

**Reply by thkl** [reply]
Es entstehen bei der Auswahl von "BOTH" zwei Schalter links der den kurzen Tastendruck auslöst, rechts der für den langen.
Siehe Beschriftung

**Comment by jb-home** [comment]
Den Status wird man darüber aber nicht sehen!
Das ist ja ein Taster/Button. Der hat in HomeMatic nur kurz einen Status. Aber nicht den vom 'Ziel' sondern nut, ob er gedrückt wurde oder nicht.
Also kein Schalter sondern ein Button. Ein Schalter kann in HomeMatic nur An und Aus. Und keinen Short oder Long Press.

**Reply by thkl** [reply]
Genau, aber da es keine Buttons in HomeKit gibt müssen es Schalter sein. Die schalten automatisch nach 500ms wieder aus, lösen aber den entsprechenden PRESS_SHORT oder PRESS_LONG Event aus.
Wenn man einen Status haben möchte, dann macht es mehr Sinn, das Gerät selber zu steuern.  Die "Taster-Schalter" sind maximal geeignet, vorhandene Taster (welche Aktionen triggern) fernauszulösen
